### PR TITLE
Benchmark_app batch calculation fix

### PR DESCRIPTION
**Pull request by fzhar**
### Details:
Fix for whole network batch calculation (if layout is not set explicitly).

### Tickets:
CVS-75101

### CODE REVIEW

**Review by vladimir-dudnik — APPROVED** [review]

File: samples/cpp/benchmark_app/main.cpp [hunk 1]
```diff
@@ -485,9 +485,10 @@ int main(int argc, char* argv[]) {
                                            });
 
             topology_name = model->get_friendly_name();
-            // use batch size according to provided layout and shapes (static case)
-            if (!isDynamicNetwork) {
-                batchSize = getModelInputBatchSize(*model);
+
+            // Calculate batch size according to provided layout and shapes (static case)
+            if (!isDynamicNetwork && app_inputs_info.size()) {
+                batchSize = getBatchSize(app_inputs_info.front());
```
Comment (nkogteva) [inline samples/cpp/benchmark_app/main.cpp:491]: Why do you get front? Maybe it is better to use maximum batch size for case where one input has batch size > 1 and others not. I'm not sure that input with batch size > 1 will be always first.

Comment (fzhar) [reply samples/cpp/benchmark_app/main.cpp:491]: This is code for static case, where each input has constant shape for every test. That's why we can use any test to calculate batch - I've taken first one (app_inputs_info is a vector of InputsInfo, each element contains info for every input, for particular test configuration).

**Comment by fzhar** [comment]
As ip/op issue was tracked in the same Jira ticket as batch issue, fix is added in the same PR

**Comment by fzhar** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
@fzhar please rebase due to changes in #9494

**Pull request merged**